plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMemoryCacheFileOperations.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultiset;
+import com.google.common.collect.Multiset;
+import io.opentelemetry.api.common.Attributes;
+import io.trino.plugin.iceberg.util.FileOperationUtils;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.filesystem.tracing.CacheSystemAttributes.CACHE_FILE_LOCATION;
+import static io.trino.plugin.iceberg.IcebergQueryRunner.ICEBERG_CATALOG;
+import static io.trino.plugin.iceberg.util.FileOperationUtils.FileType.DATA;
+import static io.trino.plugin.iceberg.util.FileOperationUtils.FileType.MANIFEST;
+import static io.trino.plugin.iceberg.util.FileOperationUtils.FileType.METADATA_JSON;
+import static io.trino.plugin.iceberg.util.FileOperationUtils.FileType.SNAPSHOT;
+import static io.trino.testing.MultisetAssertions.assertMultisetsEqual;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toCollection;
+
+@Execution(ExecutionMode.SAME_THREAD)
+public class TestIcebergMemoryCacheFileOperations
```

Comment:
I'm looking at  https://github.com/trinodb/trino/blob/eda8fdfef81e83bb81c09f3c99d76c75a70e6558/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergFileOperations.java and it feels like we're not telling the whole story here.

I'd be interested to see both `CacheOperation` and `FileOperation` accesses here.

Consider adding similar methods to:
- `testSelectWithFilter`
- `testJoin`

Reply:
Looking at FileOperation, that way it is written today, is not useful in the context of caching because it tracks `InputFile.newStream` and `InputFile.newInput` which would still be called even with caching. FileOperationUtils should instead be tracking `Input.*` calls from `TracingInput`. That still doesn't solve the problem fully because there is no corresponding `TracingInputStream` which tracks streaming reads. We instead just track creation of `TrinoInputStream` which will always happen regardless of caching.

Reply:
Added testSelectWithFilter and testJoin